ruby/type-hinting-converters-codecs-errors.md

```diff
@@ -0,0 +1,273 @@
+# Ruby SDK - Type hinting, Converters, Codecs and Errors
+
+This proposal goes into details regarding the implementation of data conversion pipeline (including
+type hints, converters and codecs) as well as outlines the hierarchy of exception classes.
+
+## Type hinting
+
+The Ruby typing system is not something we can currently rely on for specifying the types of all the
+inputs and outputs. Instead we will add custom "annotations" (technically these are just method
+calls) to the activity/workflow definitions in order to provide type hinting. Here's the basic
+example:
+
+```ruby
+class SumActivity < Temporal::Activity
+  input Integer, Integer
+  output Integer
```

Comment:
How strict is this gonna be? For example, if the args come in as two strings, are we gonna fail the workflow task? More importantly, if a workflow returns a bool, are we not gonna complete the workflow?